controllers/storagecluster/external_ocs.go

```diff
@@ -0,0 +1,221 @@
+package storagecluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	externalClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+)
+
+var (
+	// externalOCSResources will hold the ExternalResources for storageclusters
+	// ExternalResources can be accessible using the UID of an storagecluster
+	externalOCSResources = map[types.UID][]ExternalResource{}
+)
+
+// isExternalOCSProvider returns true if it is ocs to ocs ExternalStorage consumer cluster
+func isExternalOCSProvider(instance *ocsv1.StorageCluster) bool {
+	return instance.Spec.ExternalStorage.Enable && instance.Spec.ExternalStorage.StorageProviderKind == ocsv1.KindOCS
+}
+
+// newExternalClusterClient returns the *grpc.ClientConn and *providerClient.ConsumerClient
+func newExternalClusterClient(instance *ocsv1.StorageCluster) (*grpc.ClientConn, *externalClient.OCSProviderClient, error) {
+
+	// TODO: implement logic for separating address and token
+
+	grpcConnection, err := externalClient.NewGRPCConnection(
+		instance.Spec.ExternalStorage.ConnectionString, []grpc.DialOption{})
```

Comment:
pass required dial options , for insecure connection and to block until the connection is made.
```
		instance.Spec.ExternalStorage.ConnectionString, []grpc.DialOption{
			grpc.WithInsecure(),
			grpc.WithBlock(),
			grpc.WithTimeout(time.Second * 10),
		})
```

Reply:
done